### PR TITLE
Allow skipping CRC

### LINKED ISSUE #201: Allow skipping CRC
When decoding a replay, it should be possible to skip the CRC validation. It's not hugely slow, but it's largely unnecessary.

### PR DESCRIPTION
**Pull request by tfausak**
This will fix #201. For now I've done a lot of little clean up. Actually allowing the CRC to be skipped is going to be a little involved. Here's my game plan:

- Change the section type to store an un-parsed `ByteString`. In other words it's just a wrapper around a byte string with the size and CRC. 
- Barely parse the replay into the two sections. 
- If we're checking the CRC, check it. If we're not, skip it.
- Move on to parse the header.
- Used the parsed header to inform parsing the content. 
- Optional, but nice: Do something similar with the frame parsing. Barely parse it into a "section", then use the header and other contents to parse it into the full frames if appropriate. 

In other words:

1. `Replay Section Section`
2. `Replay Header Section`
3. `Replay Header (Content Section)`
4. `Replay Header (Content Frames)`

### CODE REVIEW

**Comment by tfausak** [comment]
That was surprisingly easy. 

Turns out the frames part is going to be harder. Since the stream size is part of the content, I'll have to carefully shuffle things around to avoid breaking anything. Once again #190 and #191 would be nice to have.
